mealie/repos/repository_generic.py

```diff
@@ -240,11 +243,31 @@ def pagination(self, pagination: PaginationQuery, override=None) -> PaginationBa
 
         fltr = self._filter_builder()
         q = q.filter_by(**fltr)
-
         count = q.count()
 
+        # interpret -1 as "get_all"
+        if pagination.per_page == -1:
+            pagination.per_page = count
+
+        try:
+            total_pages = ceil(count / pagination.per_page)
+
+        except ZeroDivisionError:
+            total_pages = 0
+
+        # interpret -1 as "last page"
+        if pagination.page == -1:
+            pagination.page = total_pages
+
+        # failsafe for user input error
+        if pagination.page < 1:
+            pagination.page = 1
+
         if pagination.order_by:
             if order_attr := getattr(self.model, pagination.order_by, None):
+                # queries handle uppercase and lowercase differently, which is undesirable
+                order_attr = func.lower(order_attr)
+
```

Comment:
I think this is why your PG Tests are failing. PG Stores UUIDs as Integers or Bytes ( I can't remember), so when you try to use the lower() SQL function it's causing an error. SQLite I think id just storing them as strings so it works as expected.

Reply:
Ahh okay, that makes sense. I'll do a type check on that then (which I probably should've done anyway). Thanks!